src/sonic-config-engine/minigraph.py

```diff
@@ -118,19 +119,51 @@ def parse_png(png, hname):
                 startport = link.find(str(QName(ns, "StartPort"))).text
                 bandwidth_node = link.find(str(QName(ns, "Bandwidth")))
                 bandwidth = bandwidth_node.text if bandwidth_node is not None else None
-
-                if enddevice.lower() == hname.lower():
-                    if port_alias_map.has_key(endport):
-                        endport = port_alias_map[endport]
-                    neighbors[endport] = {'name': startdevice, 'port': startport}
-                    if bandwidth:
-                        port_speeds[endport] = bandwidth
+                # Chassis internal node is used in multi-asic device or chassis minigraph
+                # where the minigraph will contain the internal asic connectivity and
+                # external neighbor information. The ChassisInternal node will be used to 
+                # determine if the link is internal to the device or chassis.
+                chassis_internal_node = link.find(str(QName(ns, "ChassisInternal")))
```

Comment:
this part of code needs further explanation. what are the scenario we are dealing with?

I can think of three:
1. backend asic (all are internal neighbors)
2. frontend asic (some connection are external neighbors, some are internal neighbors)
3. whole box. (all connection are external neighbors)

it is becoming complicated and difficult to maintain. 

My suggestion is the have a separate parse_asic_png to deal with asic png parsing. asic png does not have management port, no console. easier to understand.

Reply:
added new function as per comment